test/UnitTest/System.Web.OData.Test/OData/Query/Validators/SelectExpandQueryValidatorTest.cs

```diff
@@ -270,6 +270,71 @@ public void ValidateDoesNotThrow_IfExpansionDepthIsZero_QuerySettings()
                 () => validator.Validate(selectExpandQueryOption, new ODataValidationSettings { MaxExpansionDepth = 0 }));
         }
 
+        [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        public void ValidateDoesNotThrow_IfMaxDepthEqualsToLevels(int level)
+        {
+            // Arrange
+            string expand = string.Format("Parent($levels={0})", level);
+
+            var validator = new SelectExpandQueryValidator(new DefaultQuerySettings { EnableExpand = true });
+            var builder = new ODataConventionModelBuilder();
+            builder.EntitySet<ODataLevelsTest.LevelsEntity>("Entities");
+            IEdmModel model = builder.GetEdmModel();
+            var context = new ODataQueryContext(model, typeof(ODataLevelsTest.LevelsEntity));
+            context.RequestContainer = new MockContainer();
+            var selectExpandQueryOption = new SelectExpandQueryOption(null, expand, context);
+
+            IEdmStructuredType customerType =
+                model.SchemaElements.First(e => e.Name.Equals("LevelsEntity")) as IEdmStructuredType;
+            ModelBoundQuerySettings querySettings = new ModelBoundQuerySettings();
+            querySettings.ExpandConfigurations.Add("Parent", new ExpandConfiguration
+            {
+                ExpandType = SelectExpandType.Allowed,
+                MaxDepth = level
+            });
+            model.SetAnnotationValue(customerType, querySettings);
+
+            // Act & Assert
+            Assert.DoesNotThrow(
+                () => validator.Validate(selectExpandQueryOption, new ODataValidationSettings { MaxExpansionDepth = 0 }));
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        public void Validate_Throws_IfMaxDepthLessThanLevels(int level)
+        {
+            // Arrange
+            string expand = string.Format("Parent($levels={0})", level + 1);
+
+            var validator = new SelectExpandQueryValidator(new DefaultQuerySettings { EnableExpand = true });
+            var builder = new ODataConventionModelBuilder();
+            builder.EntitySet<ODataLevelsTest.LevelsEntity>("Entities");
+            IEdmModel model = builder.GetEdmModel();
+            var context = new ODataQueryContext(model, typeof(ODataLevelsTest.LevelsEntity));
+            context.RequestContainer = new MockContainer();
+            var selectExpandQueryOption = new SelectExpandQueryOption(null, expand, context);
+
+            IEdmStructuredType customerType =
+                model.SchemaElements.First(e => e.Name.Equals("LevelsEntity")) as IEdmStructuredType;
+            ModelBoundQuerySettings querySettings = new ModelBoundQuerySettings();
+            querySettings.ExpandConfigurations.Add("Parent", new ExpandConfiguration
```

Comment:
It would be nice to see that the annotations work when you decorate 2 properties with different settings.

Reply:
That is covered by E2E test that I tuned